### PR TITLE
Add `tvos` as a new platform

### PR DESCRIPTION
**Pull request by neonichu**

### CODE REVIEW

**Comment by neonichu** [comment]
cc @segiddins

File: lib/cocoapods-core/podfile/dsl.rb [hunk 1]
```diff
@@ -314,14 +314,15 @@ def target(name, options = {})
...
+      # The current default values are `4.3` for iOS, `10.6` for OS X, `9.0` for tvOS
+      # and `2.0` for watchOS.
       #
       # If the deployment target requires it (iOS < `4.3`), `armv6`
       # architecture will be added to `ARCHS`.
       #
       # @param    [Symbol] name
       #           the name of platform, can be either `:osx` for OS X, `:ios`
-      #           for iOS or `:watchos` for watchOS.
+      #           for iOS, `:tvos` for tvOS or `:watchos` for watchOS.
```
Comment (segiddins) [inline lib/cocoapods-core/podfile/dsl.rb]: oxford comma

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 2]
```diff
@@ -437,9 +437,9 @@ def platform
       # @return [void]
       #
       def set_platform(name, target = nil)
-        unless [:ios, :osx, :watchos].include?(name)
+        unless [:ios, :osx, :tvos, :watchos].include?(name)
           raise StandardError, "Unsupported platform `#{name}`. Platform " \
-            'must be `:ios`, `:osx`, or `:watchos`.'
+            'must be `:ios`, `:osx`, `:tvos` or `:watchos`.'
```
Comment (segiddins) [inline lib/cocoapods-core/podfile/target_definition.rb]: oxford comma

**Comment by segiddins** [comment]
:heart:

**Comment by aufflick** [comment]
FYI If you specify tvos as the platform in Podfile you get:

    [!] Invalid `Podfile` file: Unsupported platform `tvos`. Platform must be `:ios`, `:osx`, or `:watchos`.. Updating CocoaPods might fix the issue.

**Comment by orta** [comment]
@aufflick you got all the gems in [the Gemfile](https://github.com/orta/Elasticity/blob/a07cbf6fe10b7d3e44f15fe1a32c020bc95c32da/Gemfile#L3-L5)?

**Comment by aufflick** [comment]
Here's my Gemfile:

```
source 'https://rubygems.org'

gem 'cocoapods', :git => 'https://github.com/CocoaPods/CocoaPods.git', :branch => 'tvos'
gem 'xcodeproj', :git => 'https://github.com/CocoaPods/Xcodeproj.git', :branch => 'tvos-support'

gem 'cocoapods-keys', :git => 'https://github.com/orta/cocoapods-keys.git'

gem 'xcpretty'
gem 'shenzhen'
gem 'sbconstants'
```

**Comment by neonichu** [comment]
@aufflick you're missing:

```ruby
gem 'cocoapods-core', :git => 'https://github.com/CocoaPods/Core.git', :branch => 'tkos-support'
```

**Comment by aufflick** [comment]
Yes, I just noticed that - things have changed since I last built from source :)

I still see the problem though :(

```
[!] Invalid `Podfile` file: Unsupported platform `tvos`. Platform must be `:ios`, `:osx`, or `:watchos`.. Updating CocoaPods might fix the issue.

 #  from /Users/aufflick/src/_Pumptheory/MeetingPresent/MeetingPresentMac/Podfile:10
 #  -------------------------------------------
 #  target 'Foo' do
 >    platform :tvos, '9.0'
 #  
 #  -------------------------------------------
```

Gemfile:

```
source 'https://rubygems.org'

gem 'cocoapods', :git => 'https://github.com/CocoaPods/CocoaPods.git', :branch => 'tvos'
gem 'cocoapods-core', :git => 'https://github.com/CocoaPods/Core.git', :branch => 'tvos-support'
gem 'xcodeproj', :git => 'https://github.com/CocoaPods/Xcodeproj.git', :branch => 'tvos-support'

gem 'cocoapods-keys'
gem 'cocoapods-deintegrate'
```

**Comment by neonichu** [comment]
hm, not sure — I'm OSS'ing a tvOS project today, then you can compare :)

**Comment by aufflick** [comment]
Gah, this time I forgot to bundle exec - working now. Thanks @neonichu, thanks @orta.

**Comment by aufflick** [comment]
BTW: the Gemfile guide is out of date: http://guides.cocoapods.org/using/a-gemfile.html

**Comment by orta** [comment]
How so?

**Comment by segiddins** [comment]
@neonichu :shipit: with a rebase

**Comment by quentinfasquel** [comment]
So is there a proper way to already have a Podfile that specify dependencies for a :tvos target?
What's the full process? Thank you

**Comment by neonichu** [comment]
You can see an example project here: <https://github.com/contentful/tvful> — basically, you need a Gemfile, as discussed in <https://guides.cocoapods.org/using/unreleased-features>

**Comment by quentinfasquel** [comment]
Awesome thank you!

**Pull request merged**